pkg/utils/uuid_test.go

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package utils
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/heketi/tests"
+)
+
+// test actual generation of a uuid, we can only test the length
+// of the output as we are relying on an actual random source here
+func TestGenUUID(t *testing.T) {
+	uuid := GenUUID()
+	tests.Assert(t, len(uuid) == 32, "bad length", len(uuid), 32)
+}
+
+// test actual output by specifying our own source of "randomness"
+func TestFakeUUID(t *testing.T) {
+	r := strings.NewReader("heketiheketiheketi")
+	uuid := IdSource{r}.ReadUUID()
```

Comment:
Instead of using the `IdSource{r}.ReadUUID()`, wouldn't it be even nicer to have a function to initialize a new UUID generator `g = NewUuidGenerator(source)` and then later do `g.GenUUID()` ? Just a thought ...

Reply:
Perhaps I'm missing something but I'm not sure what the difference would be?